torch/nn/attention/flex_attention.py

```diff
@@ -151,18 +151,15 @@ def _ordered_to_dense(num_blocks_in_row: Tensor, col_indices: Tensor):
 
     def create_dense_one(kv_num_blocks, kv_indices):
         dense_mask = kv_indices.new_zeros(num_rows, num_cols + 1, dtype=torch.int32)
-
-        row_indices = torch.arange(num_rows, dtype=torch.int, device=device).unsqueeze(
-            -1
-        )
         col_range = torch.arange(num_cols, dtype=torch.int, device=device)
         index_mask = col_range < kv_num_blocks.unsqueeze(-1)
 
         # We write to one spot "out of bounds"
         valid_indices = torch.where(index_mask, kv_indices, num_cols)
 
         # set the values in 'a' to 1 where the indices are valid
-        dense_mask[row_indices, valid_indices] = 1
+        values = torch.ones_like(valid_indices, dtype=dense_mask.dtype)
+        dense_mask.scatter_(1, valid_indices.to(torch.int64), values)
```

Comment:
```suggestion
        dense_mask.index_fill_(1, valid_indices.to(torch.int64), 1)
```
No need to allocate a giant temporary array. Index_fill_ is graphable, no?

Reply:
Need the scatter since the indices arent vectors but you are right I can use just 1 for the fill value